api/lib/src/lib.rs

```diff
@@ -373,12 +373,21 @@ pub trait BrokerApi: SignedExtrinsicApi + StorageApi + Sized + Send + Sync + 'st
 		channel_metadata: Option<CcmChannelMetadata>,
 		boost_fee: Option<BasisPoints>,
 		affiliate_fees: Option<Affiliates<AccountId32>>,
-		refund_parameters: Option<RefundParametersRpc>,
+		refund_parameters: RefundParametersRpc,
 		dca_parameters: Option<DcaParameters>,
 	) -> Result<SwapDepositAddress> {
 		let destination_address = destination_address
 			.try_parse_to_encoded_address(destination_asset.into())
 			.map_err(anyhow::Error::msg)?;
+
+		let internal_refund_parameters = ChannelRefundParametersEncoded {
+			retry_duration: refund_parameters.retry_duration,
+			refund_address: refund_parameters
+				.refund_address
+				.try_parse_to_encoded_address(source_asset.into())
+				.map_err(anyhow::Error::msg)?,
```

Comment:
```suggestion
				.try_parse_to_encoded_address(source_asset.into())?,
```

Does this not work?

Reply:
In fact we can just use `refund_address.try_map_address(...)`? (A bit like below where we use map_address)